src/main/java/net/dv8tion/jda/client/entities/Application.java

```diff
@@ -0,0 +1,295 @@
+/*
+ *     Copyright 2015-2017 Austin Keener & Michael Ritter
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.client.entities;
+
+import java.util.Collection;
+import java.util.List;
+import net.dv8tion.jda.client.managers.ApplicationManager;
+import net.dv8tion.jda.client.managers.ApplicationManagerUpdatable;
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.Permission;
+import net.dv8tion.jda.core.entities.ISnowflake;
+import net.dv8tion.jda.core.managers.ChannelManager;
+import net.dv8tion.jda.core.requests.RestAction;
+
+/** 
+ * Represents a Discord Application from it's owning client point of view
+ * 
+ * @since  JDA 3.0
+ * @author Aljoscha Grebe
+ * 
+ * @see <a href="https://discordapp.com/developers/applications/me">Discord Documentation - My Apps</a>
+ * @see {@link net.dv8tion.jda.client.JDAClient#getApplications() JDAClient#getApplications()}
+ * @see {@link net.dv8tion.jda.client.JDAClient#getApplicationById(String) JDAClient#getApplicationById(String)}
+ */
+public interface Application extends ISnowflake
+{
+
+    /**
+     * Creates a new Bot for this Application.
+     * <b>This cannot be undone!</b>
+     * A new Bot will only be created if no bot-account is already assigned, otherwise the existing one is returned.
+     * A newly created Bot-account will have its name set to the name of the Application.
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} -
+     *         Type: {@link Application.Bot}
+     *         <br>The created bot account of this application. 
+     */
+    RestAction<Application.Bot> createBot();
```

Comment:
As this isn't part of the documanted api I don't know which responses can be returned. Same applies to the other ones. Should I just mention that it return one without specifying which ones in that case?